pynicotine/users.py

```diff
@@ -28,14 +28,16 @@
 
 class WatchedUser:
 
-    __slots__ = ("username", "upload_speed", "files", "folders")
+    __slots__ = ("username", "upload_speed", "files", "folders", "modes", "implicit")
 
     def __init__(self, username):
 
         self.username = username
         self.upload_speed = None
         self.files = None
         self.folders = None
+        self.modes = set()
+        self.implicit = True  # True if no WatchUser message was sent (i.e. receiving updates for room user)
```

Comment:
Consider a `set()` to contain all the `rooms` that a user is in, avoids doing string concatenations.

If the set is empty the user would not be in any of the rooms we are in.

Reply:
Similar concerns as the previous comment. I don't think the users module should know this.

Reply:
I'll get rid of `mode.startswith("chatroom_")` in favor of a boolean parameter for watch_user().